deeplabcut/live/dlc_live.py

```diff
@@ -32,131 +30,180 @@ class DLCLive(object):
+            try:
+                self.dlc_cfg = load_config(str(path_test_config))
+            except FileNotFoundError:
+                raise FileNotFoundError("It seems the model for shuffle {} and trainFraction {} does not exist.".format(
+                    self.shuffle, trainFraction)
+                )
+
+        else:
+            ValueError('config given at instantiation must be the config.yaml file at the root of the project directory!')
+
+
+        # gpu selection
+        if self.gputouse:
+            os.environ['CUDA_VISIBLE_DEVICES'] = str(self.gputouse)
+
+        if self.iteration:
             self.cfg['iteration'] = self.iteration
-        trainFraction = self.cfg['TrainingFraction'][self.trainingsetindex]
 
         if self.cropping is not None:
             self.cfg['cropping']=True
             self.cfg['x1'], self.cfg['x2'], self.cfg['y1'], self.cfg['y2'] = self.cropping
             print("Overwriting cropping parameters:", self.cropping)
 
-        self.cfg["project_path"] = os.path.dirname(config)
-        modelfolder = os.path.join(self.cfg["project_path"],str(auxiliaryfunctions.GetModelFolder(trainFraction, self.shuffle, self.cfg)))
-        path_test_config = Path(modelfolder) / 'test' / 'pose_cfg.yaml'
-        try:
-            self.dlc_cfg = load_config(str(path_test_config))
-        except FileNotFoundError:
-            raise FileNotFoundError("It seems the model for shuffle %s and trainFraction %s does not exist."%(self.shuffle, trainFraction))
-
         # Check which snapshots are available and sort them by # iterations
         try:
-          Snapshots = np.array([fn.split('.')[0]for fn in os.listdir(os.path.join(modelfolder , 'train'))if "index" in fn])
+            snapshots = np.array([fn.split('.')[0]for fn in os.listdir(os.path.join(self.model_folder , 'train'))if "index" in fn])
         except FileNotFoundError:
-          raise FileNotFoundError("Snapshots not found! It seems the dataset for shuffle %s has not been trained/does not exist.\n Please train it before using it to analyze videos.\n Use the function 'train_network' to train the network for shuffle %s."%(shuffle,shuffle))
+            raise FileNotFoundError("Snapshots not found! It seems the dataset for shuffle {} has not been trained/does not exist.".format(self.shuffle) +
+                                    "Please train it before using it to analyze videos." +
+                                    "Use the function 'train_network' to train the network for shuffle {}.".format(self.shuffle))
 
         if self.cfg['snapshotindex'] == 'all':
-            print("Snapshotindex is set to 'all' in the config.yaml file. Running video analysis with all snapshots is very costly! Use the function 'evaluate_network' to choose the best the snapshot. For now, changing snapshot index to -1!")
+            Warning("Snapshotindex is set to 'all' in the config.yaml file." +
```

Comment:
why is this reformated? This is a pycharm issue I'm not a fan of...

Reply:
sry, some parts of pep8 are just muscle memory at this point.
https://www.python.org/dev/peps/pep-0008/#maximum-line-length